### PR TITLE
Updated neutron configs for mitaka

### PR DESCRIPTION
**Pull request by wangxian-ibm**
Initial PR for fixing neutron configurations for migration to Mitaka.

### CODE REVIEW

**Comment by ssrikant20** [comment]
This is the initial PR minimally verified to ensure basic networking works in mitaka.

File: roles/neutron-common/templates/etc/neutron/dhcp_agent.ini [hunk 1]
```diff
@@ -1,13 +1,30 @@
 # {{ ansible_managed }}
 
 [DEFAULT]
+debug = True
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/dhcp_agent.ini]: Why would this be hardcoded to True?

File: roles/neutron-common/templates/etc/neutron/l3_agent.ini [hunk 2]
```diff
@@ -1,17 +1,19 @@
 [DEFAULT]
-# Interface driver #
+debug = False
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/l3_agent.ini]: Why is this hardcoded to false?

File: roles/neutron-common/templates/etc/neutron/dhcp_agent.ini [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+enable_isolated_metadata = True
+
+state_path = /var/lib/neutron
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/dhcp_agent.ini]: This needs to be a new variable, `neutron.state_path` similar to what we do for nova and glance.

File: roles/neutron-common/templates/etc/neutron/l3_agent.ini [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
+
+state_path = /var/lib/neutron
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/l3_agent.ini]: This should be `neutron.state_path`

File: roles/neutron-common/templates/etc/neutron/l3_agent.ini [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
 {% endif %}
 
 {% if neutron.plugin == 'ml2' %}
-# External network bridge (leave blank) #
 external_network_bridge =
 {% endif %}
 
 router_delete_namespaces = True
 
-# Layer-3 high-availability (VRRP)
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/l3_agent.ini]: Why are these comments removed?

File: roles/neutron-common/templates/etc/neutron/metadata_agent.ini [hunk 6]
```diff
@@ -1,8 +1,15 @@
 # {{ ansible_managed }}
 
 [DEFAULT]
+
+debug = True
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/metadata_agent.ini]: Why hardcoded?

File: roles/neutron-common/templates/etc/neutron/neutron.conf [hunk 7]
```diff
@@ -17,6 +17,15 @@ bind_port = {{ endpoints.neutron.port.backend_api }}
 log_dir = /var/log/neutron
 debug = {{ neutron.logging.debug }}
 
+###
+#agent_down_time = {{ neutron.agent_down_time }}
+#auth_strategy = keystone
+dhcp_agents_per_network = 2
+#report_interval = {{ neutron.report_interval }}
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/neutron.conf]: Why are we adding in commented out entries?

File: roles/neutron-common/templates/etc/neutron/neutron.conf [hunk 8]
```diff
@@ -109,6 +119,3 @@ rabbit_host = {{ endpoints.rabbit }}
 {% endif -%}
 rabbit_userid = {{ rabbitmq.user }}
 rabbit_password = {{ secrets.rabbit_password }}
-
-[oslo_middleware]
-enable_proxy_headers_parsing = True
```
Comment (j2sol) [inline roles/neutron-common/templates/etc/neutron/neutron.conf]: Why was this removed? This is needed for proxy filtering to make sure http becomes https.

Comment (wangxian-ibm) [inline roles/neutron-common/templates/etc/neutron/neutron.conf]: The reason to take out this was, https://github.com/openstack/oslo.middleware/blob/stable/mitaka/oslo_middleware/http_proxy_to_wsgi.py, stable/mitaka doesn't have it. However, I notice that it is added back in master. So should we keep it or not?

Comment (wangxian-ibm) [inline roles/neutron-common/templates/etc/neutron/neutron.conf]: Verified with Sangeetha and Niraj, this is needed. I will put it back.

**Comment by wangxian-ibm** [comment]
Making changes as suggested by j2sol, will update the PR

**Comment by j2sol** [comment]
Please rebase on master for Ansible 2.1 fixes.

**Comment by nirajdp76** [comment]
retest

**Comment by wangxian-ibm** [comment]
Jesse, rebase was done and ansible 2.1 fixes are in branch now. https://github.com/wangxian-ibm/ursula/commits/neutron But what is the "1 expected check" in the above?

**Comment by j2sol** [comment]
retest

**Comment by j2sol** [comment]
@wangxian-ibm an admin needed to trigger the CI tests, which I've done now.

**Comment by timrchavez** [comment]
:+1: I've run Ursula PR test on tardis-sl with this patch applied to a local clone and had much success.
